### PR TITLE
Adding monitoring/alerts training 6.X

### PR DESCRIPTION
**Pull request by aelkugia**
Put together monitoring/alerts section for 6.X. Updated readme to clarify original training was for 5.5 and earlier.

### CODE REVIEW

**Comment by aelkugia** [comment]
Still have a few todo's listed (related to prometheus/alertmanager commands) but pushed PR to get draft up. Some

File: monitoring-6.x.md [hunk 1]
```diff
@@ -0,0 +1,948 @@
...
+
+Most of the cluster metrics are collected by Prometheus which uses the following in-cluster services:
+
+*   [node-exporter](https://github.com/prometheus/node_exporter) (collects hardware and OS metrics)
+*   [kube-state-metrics](https://github.com/kubernetes/kube-state-metrics) (collects Kubernetes resource metrics - deployments, nodes, pods)
+
+As mentioned, node-exporter collects hardware and OS metrics. There are various collectors that are dependent on each operating system. A list of the collectors can be found [here](https://github.com/prometheus/node_exporter#collectors). By default node_exporter will expose all metrics from enabled collectors. For advanced use the node_exporter can be passed an optional list of collectors to filter metrics. In Prometheus config you can use the following syntax:
+
+```
+ params:
```
Comment (r0mant) [inline monitoring-6.x.md]: I don't think we expose a good way of modifying Prometheus configuration so I'm not sure there's value in mentioning this.

File: monitoring-6.x.md [hunk 2]
```diff
@@ -0,0 +1,948 @@
...
+
+kube-state-metrics exposes raw data that is unmodified from the Kubernetes API, which allows users to have all the data they require and perform heuristics as they see fit. In return, kubectl may not show the same values, as kubectl applies certain heuristics to display cleaner messages.
+
+Metrics from kube-state-metrics service are exported on the HTTP endpoint `/metrics` on the listening port (default 8080) and are designed to be consumed by Prometheus.
+
+![diagram](https://miro.medium.com/max/832/1*7thrW4Wa5y6b03PxtPlQzA.jpeg) 
+
+(Source: https://medium.com/faun/production-grade-kubernetes-monitoring-using-prometheus-78144b835b60)
+
+All metrics collected by node-exporter and kube-state-metrics are placed into the k8s database in Prometheus. See below for a list of metrics collected by Prometheus. Each metric is stored as a separate “series” in Prometheus.
```
Comment (r0mant) [inline monitoring-6.x.md]: There is no concept of "database" in Prometheus as far as I'm aware. It just stores a bunch of time series. https://prometheus.io/docs/concepts/data_model/

File: monitoring-6.x.md [hunk 3]
```diff
@@ -0,0 +1,948 @@
...
+Like mentioned above, Prometheus is exposed via a cluster-local Kubernetes service `prometheus-k8s.monitoring.svc.cluster.local:9090` and serves its HTTP API on port `9090` so we can use it to explore the database from the CLI.
+
+Also, as seen above we have the following Prometheus pods:
+
+```
+prometheus-adapter-6586cf7b4f-hmwkf   
+prometheus-k8s-0                       
+prometheus-operator-7bd7d57788-mf8xn   
+```
+Prometheus operator for kuebrnetes allows easy monitoring defintions for kubernetes services and deployment and managment of Prometheus instances.
```
Comment (r0mant) [inline monitoring-6.x.md]: kuebrnetes -> Kubernetes, defintions -> definitions, managment -> management

File: monitoring-6.x.md [hunk 4]
```diff
@@ -0,0 +1,948 @@
...
+
+Refer to the Prometheus [API documentation](https://prometheus.io/docs/prometheus/latest/querying/basics/) if you want to learn more about querying the database.
+
+## Metric Retention Policy 
+
+### Time based retention
+
+Users should start Prometheus with the following flag with their desired time.
+
+`--storage.tsdb.retention.time`: Determines when to remove old data. Defaults to 15 days.
```
Comment (r0mant) [inline monitoring-6.x.md]: Like above, not sure we should be showing CLI flags as we're configuring Prometheus ourselves and this is not exposed to the user.

What we should mention, however, is our retention period. It's 30 days IIRC. cc @UnderGreen right?

File: monitoring-6.x.md [hunk 5]
```diff
@@ -0,0 +1,948 @@
...
+
+Following Units are supported: KB, MB, GB, PB. for Ex: "512MB"
+
+We can use the same Prometheus API to see the retention policies configured for `--storage.tsdb.retention.time`:
+
+```bash
+$ curl http://prometheus-k8s.monitoring.svc.cluster.local:9090/api/v1/status/flags | jq
+```
+
+The configuration of retention policies and rollups are handled by a “watcher” pod so all these configurations can be seen in its logs:
```
Comment (r0mant) [inline monitoring-6.x.md]: Retention policies are not configurable with Prometheus IIRC.

File: monitoring-6.x.md [hunk 6]
```diff
@@ -0,0 +1,948 @@
...
+  </tr>
+  <tr>
+   <td>uptime (integer, seconds)
+   </td>
+   <td>Number of milliseconds since the system was started
+   </td>
+  </tr>
+</table>
+
+In addition to the default metrics, Telegraf also queries the Satellite Prometheus endpoint described above and ships all metrics to the same “k8s” database in Prometheus.
```
Comment (r0mant) [inline monitoring-6.x.md]: There's no "k8s" database in Prometheus.

File: monitoring-6.x.md [hunk 7]
```diff
@@ -0,0 +1,948 @@
...
+```
+$ gravity resource rm alerttarget email-alerts
+```
+
+### Testing Alertmanager Email Configuration
+
+To test a Alertmanager SMTP configuration you can execute the following:
+
+```
+$ kubectl exec -n monitoring $POD_ID -c alertmanager -- /bin/bash -c "alertmanager service-tests smtp"
```
Comment (r0mant) [inline monitoring-6.x.md]: Isn't that Kapacitor's smtp test command?

File: monitoring-6.x.md [hunk 8]
```diff
@@ -0,0 +1,948 @@
...
+
+For demonstration purposes let’s define an alert that always fires:
+
+```
+kind: alert
+version: v2
+metadata:
+  name: my-formula
+spec:
+  formula: |
```
Comment (r0mant) [inline monitoring-6.x.md]: This is Kapacitor's alert definition, it won't work with Alertmanager.

Comment (r0mant) [reply monitoring-6.x.md]: I have an example of a custom Alertmanager alert that always fires:

```yaml
kind: alert
version: v2
metadata:
  name: cpu1
spec:
  alert_name: CPU1
  group_name: test-group
  formula: |
    node:cluster_cpu_utilization:ratio * 100 > 1
  labels:
    severity: info
  annotations:
    description: |
      This is a test alert
```

File: monitoring-6.x.md (hunk 7 above)
Comment (aelkugia) [reply monitoring-6.x.md]: Removed for now, I'll see if I can find a way to test so we can show during the training

File: monitoring-6.x.md (hunk 8 above)
Comment (aelkugia) [reply monitoring-6.x.md]: Sounds good, update to this custom alert and following commands reflect this

File: monitoring-6.x.md (hunk 4 above)
Comment (aelkugia) [reply monitoring-6.x.md]: Updated to 30 days

**Pull request merged**